packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -494,16 +494,19 @@ class CupertinoSheetRoute<T> extends PageRoute<T> with _CupertinoSheetRouteTrans
 
   @override
   Widget buildContent(BuildContext context) {
-    final double bottomPadding = MediaQuery.sizeOf(context).height * _kTopGapRatio;
+    final double topPadding = MediaQuery.sizeOf(context).height * _kTopGapRatio;
     return MediaQuery.removePadding(
       context: context,
       removeTop: true,
       removeBottom: true,
       child: Padding(
-        padding: EdgeInsets.only(bottom: bottomPadding),
-        child: CupertinoUserInterfaceLevel(
-          data: CupertinoUserInterfaceLevelData.elevated,
-          child: _CupertinoSheetScope(child: builder(context)),
+        padding: EdgeInsets.only(top: topPadding),
+        child: ClipRRect(
+          borderRadius: const BorderRadius.vertical(top: Radius.circular(12)),
```

Comment:
I think this makes the ClipRRect usage in the build method and  of CupertinoSheetTransition and _coverSheetSecondaryTransition right? Can we remove those?

Reply:
Correct me if i am wrong but any `ClipRRect` above `Padding` will add padding first then Clip the child. But here we want to Clip first and then add padding right.

Because any ClipRRect above this will just clip the transparent part which is remaining at top after padding.

Reply:
Apologies, I had a typo in my comment that I think made things confusing.

Currently the rounded corners are added by two `ClipRRect`s in CupertinoSheetTransition, [here](https://github.com/flutter/flutter/blob/a0b1b3253416b1f19e3f105b596eb6e4c5813ef4/packages/flutter/lib/src/cupertino/sheet.dart#L426C9-L426C26) and [here](https://github.com/flutter/flutter/blob/a0b1b3253416b1f19e3f105b596eb6e4c5813ef4/packages/flutter/lib/src/cupertino/sheet.dart#L443), which both wrap around this code. With the ClipRRect being added here, I think it makes those unneeded?

Reply:
Yes, make sense. Pushing the change.

Reply:
Have pushed the required change. Please have a look.